app/client/ui/UserManager.ts

```diff
@@ -233,6 +239,19 @@ export class UserManager extends Disposable {
     ];
   }
 
+  private _onAddOrEdit(email: string, role: roles.NonGuestRole) {
+    let found = null;
+    computedArray(this._model.membersEdited, member => {
+      if (member.email === email) {
+        member.access.set(role);
+        found = true;
+      }
+    });
+    if (!found) {
+      this._onAdd(email, role);
+    }
```

Comment:
```suggestion
    const members = this._model.membersEdited.get();
    const maybeMember = members.find(m => m.email === email);
    if (maybeMember) {
      maybeMember.access.set(role);
    } else {
      this._onAdd(email, role);
    }
```

To avoid using `computedArray`.